packages/react/src/components/DataGrid/utils.tsx

```diff
@@ -46,15 +48,24 @@ export const useHeadlessColumns = <T extends Row>({
   if (enableRowSelection) {
     headlessColumns = [
       columnHelper.display({
-        id: "select",
-        header: ({ table }) => (
-          <Checkbox
-            checked={table.getIsAllRowsSelected()}
-            indeterminate={table.getIsSomeRowsSelected()}
-            onChange={table.getToggleAllRowsSelectedHandler()}
-            aria-label={t("components.datagrid.rows_selection_aria")}
-          />
-        ),
+        id: HEADER_ID_SELECT,
+        header: ({ table }) => {
+          const render = useMemo(() => {
+            return (
+              <Checkbox
+                checked={table.getIsAllRowsSelected()}
+                indeterminate={table.getIsSomeRowsSelected()}
+                onChange={table.getToggleAllRowsSelectedHandler()}
+                aria-label={t("components.datagrid.rows_selection_aria")}
+              />
+            );
+          }, [
+            table.getIsAllRowsSelected(),
+            table.getIsSomeRowsSelected(),
+            table.getToggleAllRowsSelectedHandler(),
+          ]);
+          return render;
```

Comment:
Since you are rendering Checkbox directly, it seems this render method is useless nope ?

Reply:
tbh I can't remember why I did this 🤔